

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Quality control and feature tabulation &mdash; QIIME2-workshop 2021.06 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Taxonomic classification" href="taxonomy.html" />
    <link rel="prev" title="Getting started" href="gettingstarted.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> QIIME2-workshop
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="genintro.html">General Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Quality control and feature tabulation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#prepping-the-manifest-file">Prepping the manifest file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#primer-removal">Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="#feature-table-construction">Feature table construction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#deblur-denoise">Deblur denoise</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#step-1-join-read-pairs">Step 1. Join read pairs</a></li>
<li class="toctree-l4"><a class="reference internal" href="#step-2-quality-filter">Step 2. Quality filter</a></li>
<li class="toctree-l4"><a class="reference internal" href="#step-3-denoise">Step 3. Denoise</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#dada2-denoise">DADA2 denoise</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="taxonomy.html">Taxonomic classification</a></li>
<li class="toctree-l1"><a class="reference internal" href="crunchomics.html">Metabarcoding analyses on Crunchomics</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">QIIME2-workshop</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Quality control and feature tabulation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="quality-control-and-feature-tabulation">
<h1>Quality control and feature tabulation<a class="headerlink" href="#quality-control-and-feature-tabulation" title="Permalink to this headline">¶</a></h1>
<p>There are many ways in which to enter the QIIME2 workflow. Which one to use depends mostly on whether your data has been preprocessed in any way.
In our case, sequences were already demultiplexed by the sequencing facility, such that we have two files for each of the samples:
one containing the forward read “R1” and one with the reverse “R2”.
We can import all these sequence files into QIIME2 in one go using a manifest file.</p>
<p>See the QIIME2 <a class="reference external" href="https://docs.qiime2.org/2020.8/tutorials/importing/">Importing data</a> tutorial information on how to import different input data formats.</p>
<div class="section" id="prepping-the-manifest-file">
<h2>Prepping the manifest file<a class="headerlink" href="#prepping-the-manifest-file" title="Permalink to this headline">¶</a></h2>
<p>The manifest file includes the sample ids, the path to where each fastq.gz file is stored, and its orientation.
Because we have paired-end data and thus two files for each sample, we will list each sample twice, once for the forward and once for the reverse orientation.
This is what the first few lines of the manifest file look like (or run <code class="docutils literal notranslate"><span class="pre">head</span> <span class="pre">data/MANIFEST.csv</span></code>):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sample-id,absolute-filepath,direction
DUMMY10,<span class="nv">$PWD</span>/data/Sample-DUMMY10_R1.fastq.gz,forward
DUMMY10,<span class="nv">$PWD</span>/data/Sample-DUMMY10_R2.fastq.gz,reverse
DUMMY11,<span class="nv">$PWD</span>/data/Sample-DUMMY11_R1.fastq.gz,forward
DUMMY11,<span class="nv">$PWD</span>/data/Sample-DUMMY11_R2.fastq.gz,reverse
...
</pre></div>
</div>
<p>You can find the manifest file under data/MANIFEST.csv as a comma-separated file.</p>
</div>
<div class="section" id="id1">
<h2>Importing data<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Don’t forget to activate your conda environment before running any QIIME2 command.
Just run: <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">activate</span> <span class="pre">qiime2-2021.2</span></code>.</p>
</div>
<p>We can use the manifest file to import the data into QIIME2.
I already selected the appropriate data type and input format, so just run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir -p prep
qiime tools import <span class="se">\</span>
  --type <span class="s1">&#39;SampleData[PairedEndSequencesWithQuality]&#39;</span> <span class="se">\</span>
  --input-path data/MANIFEST.csv <span class="se">\</span>
  --input-format PairedEndFastqManifestPhred33 <span class="se">\</span>
  --output-path prep/demux-seqs.qza
</pre></div>
</div>
<p>This will create a special data object called an ‘artifact’ which is a zip archive with a ‘.qza’ (QIIME zipped artifact) extension.
Artifacts are the main file type used in QIIME2 analyses.
In addition to the actual data, in this case the fastq.gz files,
artifacts contain metadata such as file format, relevant citations and the data provenance.</p>
<p>If you quickly want to see what sort of artifact it is, run</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime tools peek prep/demux-seqs.qza
</pre></div>
</div>
<p>From the demultiplexed artifact we can create an interactive summary of the sequences.
This summary includes the number of sequences that were obtained per sample and the distribution of sequence quality scores at each position.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime demux summarize <span class="se">\</span>
  --i-data prep/demux-seqs.qza <span class="se">\</span>
  --o-visualization prep/demux-seqs.qzv
</pre></div>
</div>
<p>The result is the other main object in QIIME2: a ‘vizualisation’, which is a zip archive with ‘.qzv’ (QIIME zipped vizualisation) extension.
Unlike artifacts, <strong>vizualisations cannot be used as input for further analyses in QIIME2</strong>.
They are only ment for display, for instance using the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code> utility.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime tools view prep/demux-seqs.qzv
</pre></div>
</div>
<p>Alternatively, we can view it online: <a class="reference external" href="https://view.qiime2.org/">https://view.qiime2.org/</a>.
This online utility is also handy to share QIIME2 results with collaborators that do not have QIIME2 installed.</p>
<div class="admonition-question-1 admonition">
<p class="admonition-title">Question 1</p>
<p>Check out the interactive quality plots. What could cause the ‘drop’ in quality at the 5-prime end of your sequences?</p>
</div>
</div>
<div class="section" id="primer-removal">
<h2>Primer removal<a class="headerlink" href="#primer-removal" title="Permalink to this headline">¶</a></h2>
<p>The next step is to remove any primer sequences. We will use the cutadapt QIIME2 plugin for that.
Because we have paired-end data, there is a forward and reverse primer, referenced by the parameters <code class="docutils literal notranslate"><span class="pre">--p-front-f</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-front-r</span></code> in below command.
These are already the correct primer sequences used for your data.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Some steps in the QIIME2 workflow can take quite some time.
We can speed things up by running certain processes using more than one cpu.
Exactly how many cpus you can select mainly depends on how many you have available on your computer.</p>
</div>
<p>For the following command, I used <code class="docutils literal notranslate"><span class="pre">--p-cores</span> <span class="pre">8</span></code> (but please adjust the number of cpus based on how many you have available!).</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime cutadapt trim-paired <span class="se">\</span>
  --i-demultiplexed-sequences prep/demux-seqs.qza <span class="se">\</span>
  --p-front-f GTGYCAGCMGCCGCGGTAA <span class="se">\</span>
  --p-front-r CCGYCAATTYMTTTRAGTTT <span class="se">\</span>
  --p-error-rate <span class="m">0</span> <span class="se">\</span>
  --o-trimmed-sequences prep/trimmed-seqs.qza <span class="se">\</span>
  --p-cores <span class="m">1</span> <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p>The output written to the screen shows the commands that are run, as well as some info on the number of reads that are processed and trimmed.</p>
<p>The actual result is not written to the screen but saved as another QIIME2 artifact, which contains the trimmed sequences.
Like with prep/demux-seqs.qza from the previous step, we can create a summary vizualisation and view it like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime demux summarize <span class="se">\</span>
  --i-data prep/trimmed-seqs.qza <span class="se">\</span>
  --o-visualization prep/trimmed-seqs.qzv

qiime tools view prep/trimmed-seqs.qzv
</pre></div>
</div>
<p>Compare the summary vizualisations of your demux and trimmed sequences.</p>
<div class="admonition-question-2 admonition">
<p class="admonition-title">Question 2</p>
<div class="line-block">
<div class="line">What is the effect on the 5-prime quality score?</div>
<div class="line">What is the effect on the sequence length summary?</div>
</div>
</div>
</div>
<div class="section" id="feature-table-construction">
<h2>Feature table construction<a class="headerlink" href="#feature-table-construction" title="Permalink to this headline">¶</a></h2>
<p>Sequences were traditionally clustered into operational taxonomic units based on a fixed similarity threshold, typically 97%
(<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/25177538/">Rideout et al. 2014</a>).
Such OTU clustering methods have been largely replaced now by denoise algorithms, which correct amplicon sequence errors and
produce high-resolution amplicon sequence variants that resolve differences of as little as one nucleotide
(<a class="reference external" href="https://www.nature.com/articles/ismej2017119">Callahan et al. 2017</a>).</p>
<p>Two denoisers are implemented in QIIME2: deblur-denoise and DADA2-denoise.
Their performance is quite similar so which one you use in the end depends largely on taste
(<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30123705/">Nearing et al. 2018</a>,
<a class="reference external" href="https://msystems.asm.org/content/4/1/e00163-18">Caruso et al. 2019</a>).
In the following sections you will run both.</p>
<p>See <a class="reference external" href="https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpbi.100">Estaki et al. 2020</a> for further details on the denoising workflow.</p>
<div class="section" id="deblur-denoise">
<h3>Deblur denoise<a class="headerlink" href="#deblur-denoise" title="Permalink to this headline">¶</a></h3>
<p>The Deblur-denoising procedure is split up in 3 steps.</p>
<div class="section" id="step-1-join-read-pairs">
<h4>Step 1. Join read pairs<a class="headerlink" href="#step-1-join-read-pairs" title="Permalink to this headline">¶</a></h4>
<p>First, join the forward and reverse reads into a single sequence spanning the entire target region.
This joining is based on the overlap between the forward and reverse reads.</p>
<p>Lets first do some math:</p>
<div class="admonition-question-3 admonition">
<p class="admonition-title">Question 3</p>
<div class="line-block">
<div class="line">a. What is the length of your trimmed forward and reverse reads? (Tip: check out the length of the primer sequences used in the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">cutadapt</span> <span class="pre">trim-paired</span></code> command)</div>
<div class="line">b. What is the length of the fragment that was sequenced? (Tip: the primers that were used were 515F-926R)</div>
<div class="line">c. What was the length of the primer trimmed fragment (i.e. target region)?</div>
<div class="line">d. Given the above, how much overlap do we have between your forward and reverse reads?</div>
<div class="line">e. Can you confirm the numbers in question a-b by viewing the appropriate vizualisations?</div>
</div>
</div>
<p>Now that you have checked whether there is enough overlap to reliably join your forward and reverse reads, lets do the joining.
Note that I used <cite>–p-threads 8</cite>, but you need to decide for yourself how many cpus you have available.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir -p deblur
qiime vsearch join-pairs <span class="se">\</span>
  --i-demultiplexed-seqs prep/trimmed-seqs.qza <span class="se">\</span>
  --o-joined-sequences deblur/joined-seqs.qza <span class="se">\</span>
  --p-threads <span class="m">1</span> <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p>Lets summarize and view again:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime demux summarize <span class="se">\</span>
  --i-data deblur/joined-seqs.qza <span class="se">\</span>
  --o-visualization deblur/joined-seqs.qzv

qiime tools view deblur/joined-seqs.qzv
</pre></div>
</div>
<div class="admonition-question-4 admonition">
<p class="admonition-title">Question 4</p>
<div class="line-block">
<div class="line">Can you confirm the target region length you computed in question 3c.?</div>
<div class="line">What happends to the quality in the middle of the joined reads and why?</div>
</div>
</div>
</div>
<div class="section" id="step-2-quality-filter">
<h4>Step 2. Quality filter<a class="headerlink" href="#step-2-quality-filter" title="Permalink to this headline">¶</a></h4>
<p>The fastq.gz files not only contain the DNA sequences but also a quality score for each of the nucleotides.
These quality scores are used by deblur-denoise to apply an initial quality filtering.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>With qiime2-2021.2, the following command will produce a <code class="docutils literal notranslate"><span class="pre">YAMLLoadWarning</span></code> warning, which you can ignore.</p>
</div>
<p>To perform quality filtering, just run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime quality-filter q-score <span class="se">\</span>
  --i-demux deblur/joined-seqs.qza <span class="se">\</span>
  --o-filtered-sequences deblur/filt-seqs.qza <span class="se">\</span>
  --o-filter-stats deblur/filt-stats.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p>View the summary of your joined and filtered sequences:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime demux summarize <span class="se">\</span>
  --i-data deblur/filt-seqs.qza <span class="se">\</span>
  --o-visualization deblur/filt-seqs.qzv

qiime tools view deblur/filt-seqs.qzv
</pre></div>
</div>
<p>These results actually look very, very similar to the summary vizualisation of the joined sequences.
In fact, when you look at the trimming stats, nothing really seems to have happened:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
   --m-input-file deblur/filt-stats.qza <span class="se">\</span>
   --o-visualization deblur/filt-stats.qzv

qiime tools view deblur/filt-stats.qzv
</pre></div>
</div>
<p>This is somewhat unusual but so is the quality of this particular data set. Just be prepared that your own data may be of lower quality.</p>
<div class="admonition-question-5 admonition">
<p class="admonition-title">Question 5</p>
<p>Have a look at the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">quality-filter</span> <span class="pre">q-score</span></code> help function and see if you can find an explanation why no quality filtering was performed:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime quality-filter q-score --help
</pre></div>
</div>
</div>
<p>Now lets prepare for the next step where the actual denoising is performed:</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>When we continue with deblur-denoise in the next step, you need to truncate your fragments such that they are all of the same length.
The position at which sequences are truncated is specified by the <code class="docutils literal notranslate"><span class="pre">--p-trim-length</span></code> parameter.
Any sequence that is shorter than this value will be lost from your analyses.
Any sequence that is longer will be truncated at this position.</p>
</div>
<div class="admonition-question-6 admonition">
<p class="admonition-title">Question 6</p>
<div class="line-block">
<div class="line">What happends to the quality of the fragments that are longer than the presumed fragment size? Why?</div>
<div class="line">What is an appropriate value for <code class="docutils literal notranslate"><span class="pre">--p-trim-length</span></code>?</div>
<div class="line">What proportion of your fragments will be retained approximately using this value?</div>
<div class="line">How many bases will be trimmed off from fragments of median length?</div>
</div>
</div>
</div>
<div class="section" id="step-3-denoise">
<h4>Step 3. Denoise<a class="headerlink" href="#step-3-denoise" title="Permalink to this headline">¶</a></h4>
<p>Run deblur-denoise on the joined and quality trimmed sequences.
I selected a <code class="docutils literal notranslate"><span class="pre">--p-trim-length</span></code> of 370, because that resulted in minimal data loss.
That is, only &lt;9% of the reads were discarded for being too short, and only 4 bases were trimmed off from sequences of median length.
Again, I used 8 cpus here, but you may have to modify that (<code class="docutils literal notranslate"><span class="pre">--p-jobs-to-start</span> <span class="pre">8</span></code>)</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime deblur denoise-16S <span class="se">\</span>
  --i-demultiplexed-seqs deblur/filt-seqs.qza <span class="se">\</span>
  --p-trim-length <span class="m">370</span> <span class="se">\</span>
  --o-representative-sequences deblur/deblur-reprseqs.qza <span class="se">\</span>
  --o-table deblur/deblur-table.qza <span class="se">\</span>
  --p-sample-stats <span class="se">\</span>
  --o-stats deblur/deblur-stats.qza <span class="se">\</span>
  --p-jobs-to-start <span class="m">1</span> <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p>This command results in three output files:</p>
<ol class="arabic simple">
<li><p>A feature table artifact with the frequencies per sample and feature.</p></li>
<li><p>A representative sequences artifact with one single fasta sequence for each of the features in the feature table.</p></li>
<li><p>A stats artifact with details of how many reads passed each filtering step of the deblur procedure.</p></li>
</ol>
<p>Let start with converting the stats artifact into a vizualisations which we can then view again, like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime deblur visualize-stats <span class="se">\</span>
  --i-deblur-stats deblur/deblur-stats.qza <span class="se">\</span>
  --o-visualization deblur/deblur-stats.qzv

qiime tools view deblur/deblur-stats.qzv
</pre></div>
</div>
<p>The resulting table shows how many sequences per sample passed the deblur quality check, including the total number and the number of unique sequences.
Just hoover over the table headers to get more information. Note that artifact here is used in the traditional sense of the word.
It has nothing to do with the file type ‘artifact’.</p>
<p>To view the feature table summary statistics, just run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table summarize <span class="se">\</span>
  --i-table deblur/deblur-table.qza <span class="se">\</span>
  --o-visualization deblur/deblur-table.qzv

qiime tools view deblur/deblur-table.qzv
</pre></div>
</div>
<p>Particularly interesting is the <cite>Interactive sample detail page</cite>.
This shows you the number of features per sample.
If there is large variation in this number between samples, it means it is difficult to directly compare samples. In that case it is often recommended to standardize your data by for instance rarefaction.
Rarefaction will not be treated in detail in the tutorial, but the idea is laid out below.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Rarefaction in a nut shell: for each sample you only include as many features as you have available for the smallest sample.
Which features those are is determined by chance (i.e. subsample the features without replacement).
If your smallest sample has very few features, you need to discard a lot of features from the other samples, which is of course a pity.
In such situations it may be better to just exclude the small sample entirely.</p>
</div>
<p>Use the sample depth slider to see the effect of rarefying the data up to a minimum sampling depth.</p>
<div class="admonition-question-7 admonition">
<p class="admonition-title">Question 7</p>
<div class="line-block">
<div class="line">What would be the rarefaction depth if your want to retain all your samples?</div>
<div class="line">How many features (ASVs) would you discard at this rarefaction depth?</div>
<div class="line">Is there a better option given the data? What would then be the consequence for the number of features and samples retained?</div>
</div>
</div>
<p>Lastly, lets check out the representative sequences:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data deblur/deblur-reprseqs.qza <span class="se">\</span>
  --o-visualization deblur/deblur-reprseqs.qzv

qiime tools view deblur/deblur-reprseqs.qzv
</pre></div>
</div>
<p>In addition to some summary statistics, this vizualization allows you to BLAST each representative sequence against the NCBI nt database.
Just click on the sequence and then the View report button.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Results of the “top hits” from a simple BLAST search such as this are known to be poor predictors of the true taxonomic affiliations of these features,
especially in cases where the closest reference sequence in the database is not very similar to the sequence that you are using as a query.
Instead, use automated taxonomic classification (see tutorial for tomorrow).</p>
</div>
<div class="admonition-question-8 admonition">
<p class="admonition-title">Question 8</p>
<p>What is the best BLAST hit for the most abundant feature in the data set?</p>
</div>
</div>
</div>
<div class="section" id="dada2-denoise">
<h3>DADA2 denoise<a class="headerlink" href="#dada2-denoise" title="Permalink to this headline">¶</a></h3>
<p>The DADA2-denoiser can be run using a single command because joining and filtering will be done automatically.
However, you need to decide on two important parameter values: <code class="docutils literal notranslate"><span class="pre">--p-trunc-len-f</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-trunc-len-r</span></code>.</p>
<div class="admonition-question-9 admonition">
<p class="admonition-title">Question 9</p>
<p>Check out the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">dada2</span> <span class="pre">denoise-paired</span></code> help function to find out what <code class="docutils literal notranslate"><span class="pre">--p-trunc-len-f</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-trunc-len-r</span></code> are.
Which of the previous vizualisations can help you decide on appropriate values for these parameters?
What do you think happens if you set these values at, say, 185 each?</p>
</div>
<p>Lets run dada2, note that I used 16 cpus, which you may have to tune down a bit.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime dada2 denoise-paired <span class="se">\</span>
  --i-demultiplexed-seqs prep/trimmed-seqs.qza <span class="se">\</span>
  --p-trim-left-f <span class="m">0</span> <span class="se">\</span>
  --p-trim-left-r <span class="m">0</span> <span class="se">\</span>
  --p-trunc-len-f <span class="m">230</span> <span class="se">\</span>
  --p-trunc-len-r <span class="m">220</span> <span class="se">\</span>
  --p-n-threads <span class="m">1</span> <span class="se">\</span>
  --o-table dada2/dada2-table.qza <span class="se">\</span>
  --o-representative-sequences dada2/dada2-reprseqs.qza <span class="se">\</span>
  --o-denoising-stats dada2/dada2-stats.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p>This will take some time, so lets keep it running and pick this up again this afternoon or tomorrow morning.
Then create some vizualisations:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
  --m-input-file dada2/dada2-stats.qza <span class="se">\</span>
  --o-visualization dada2/dada2-stats.qzv

qiime feature-table summarize <span class="se">\</span>
  --i-table dada2/dada2-table.qza <span class="se">\</span>
  --o-visualization dada2/dada2-table.qzv

qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data dada2/dada2-reprseqs.qza <span class="se">\</span>
  --o-visualization dada2/dada2-reprseqs.qzv
</pre></div>
</div>
<p>View and compare these results to the ones obtained with deblur-denoise.</p>
<div class="admonition-question-10 admonition">
<p class="admonition-title">Question 10</p>
<p>What method yields most features?
Which method would you choose to continue with?</p>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="taxonomy.html" class="btn btn-neutral float-right" title="Taxonomic classification" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="gettingstarted.html" class="btn btn-neutral float-left" title="Getting started" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, Evelien Jongepier

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>