<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Parameter reference &mdash; NanoClass pre-release documentation</title><link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Databases" href="databases.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> NanoClass
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="objectives.html">NanoClass objectives</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="typical_run.html">Two tools in one</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Databases</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Parameter reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#the-config-yaml">The config yaml</a></li>
<li class="toctree-l2"><a class="reference internal" href="#all-parameters-in-nanoclass">All parameters in NanoClass</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#porechop">porechop</a></li>
<li class="toctree-l3"><a class="reference internal" href="#nanofilt">nanofilt</a></li>
<li class="toctree-l3"><a class="reference internal" href="#subsample">subsample</a></li>
<li class="toctree-l3"><a class="reference internal" href="#nanostats">nanostats</a></li>
<li class="toctree-l3"><a class="reference internal" href="#nanoplot">nanoplot</a></li>
<li class="toctree-l3"><a class="reference internal" href="#common">common</a></li>
<li class="toctree-l3"><a class="reference internal" href="#blastn">blastn</a></li>
<li class="toctree-l3"><a class="reference internal" href="#dcmegablast">dcmegablast</a></li>
<li class="toctree-l3"><a class="reference internal" href="#centrifuge">centrifuge</a></li>
<li class="toctree-l3"><a class="reference internal" href="#idtaxa">idtaxa</a></li>
<li class="toctree-l3"><a class="reference internal" href="#kraken">kraken</a></li>
<li class="toctree-l3"><a class="reference internal" href="#megablast">megablast</a></li>
<li class="toctree-l3"><a class="reference internal" href="#minimap">minimap</a></li>
<li class="toctree-l3"><a class="reference internal" href="#mothur">mothur</a></li>
<li class="toctree-l3"><a class="reference internal" href="#qiime">qiime</a></li>
<li class="toctree-l3"><a class="reference internal" href="#rdp">rdp</a></li>
<li class="toctree-l3"><a class="reference internal" href="#spingo">spingo</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">NanoClass</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Parameter reference</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/parameter_ref.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="parameter-reference">
<h1>Parameter reference<a class="headerlink" href="#parameter-reference" title="Permalink to this headline"></a></h1>
<div class="section" id="the-config-yaml">
<h2>The config yaml<a class="headerlink" href="#the-config-yaml" title="Permalink to this headline"></a></h2>
<p>Parameter setting can be customized using the <cite>config.yaml</cite>.
Parameters that are commonly altered are listed in the <cite>config.yaml</cite> in the NanoClass directory, which looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>samples:                           <span class="s2">&quot;samples.csv&quot;</span>
methods:                           <span class="o">[</span><span class="s2">&quot;blastn&quot;</span>,<span class="s2">&quot;centrifuge&quot;</span>,<span class="s2">&quot;dcmegablast&quot;</span>,<span class="s2">&quot;idtaxa&quot;</span>,<span class="s2">&quot;kraken&quot;</span>,<span class="s2">&quot;megablast&quot;</span>,<span class="s2">&quot;minimap&quot;</span>,<span class="s2">&quot;mothur&quot;</span>,<span class="s2">&quot;qiime&quot;</span>,<span class="s2">&quot;rdp&quot;</span>,<span class="s2">&quot;spingo&quot;</span><span class="o">]</span>

porechop:
    checkreads:                    <span class="m">20000</span>

nanofilt:
    minlen:                        <span class="m">1400</span>
    maxlen:                        <span class="m">1600</span>
    quality:                       <span class="m">10</span>

subsample:
    skip:                          <span class="nb">false</span>
    samplesize:                    <span class="m">100</span>

common:
    dburl:                         <span class="s2">&quot;https://www.arb-silva.de/fileadmin/silva_databases/qiime/Silva_132_release.zip&quot;</span>
    ssu:                           <span class="s2">&quot;16S&quot;</span> <span class="c1">#16S or 18S</span>
    group-by:                      sample

blastn:
    lcaconsensus:                  <span class="m">0</span>.5
    evalue:                        <span class="m">0</span>.00001
    pctidentity:                   <span class="m">80</span>
    alnlength:                     <span class="m">100</span>
    ntargetseqs:                   <span class="m">50</span>

dcmegablast:
    lcaconsensus:                  <span class="m">0</span>.5
    evalue:                        <span class="m">0</span>.00001
    pctidentity:                   <span class="m">80</span>
    alnlength:                     <span class="m">100</span>
    ntargetseqs:                   <span class="m">50</span>

centrifuge:
    taxmapurl:                     <span class="s2">&quot;https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/taxonomy/taxmap_embl_ssu_ref_nr99_132.txt.gz&quot;</span>
    sequrl:                        <span class="s2">&quot;https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/SILVA_132_SSURef_Nr99_tax_silva.fasta.gz&quot;</span>

idtaxa:
    pctthreshold:                  <span class="m">60</span>

kraken:
    dbtype:                        <span class="s2">&quot;silva&quot;</span>

megablast:
    lcaconsensus:                  <span class="m">0</span>.5
    evalue:                        <span class="m">0</span>.00001
    pctidentity:                   <span class="m">80</span>
    alnlength:                     <span class="m">100</span>
    ntargetseqs:                   <span class="m">50</span>

minimap:
    lcaconsensus:                  <span class="m">0</span>.5
    ntargetseqs:                   <span class="m">10</span>
</pre></div>
</div>
<p>The indentation in the config.yaml is important. Note that tabs are not allowed, only spaces.</p>
<p>Here is a description of all tunable parameters, including more advanced usage:</p>
</div>
<div class="section" id="all-parameters-in-nanoclass">
<h2>All parameters in NanoClass<a class="headerlink" href="#all-parameters-in-nanoclass" title="Permalink to this headline"></a></h2>
<dl class="simple">
<dt><em>samples</em> [string] (default: samples.csv)</dt><dd><p>Comma separated file with run label, sample label and path to input file in fastq.gz format.
File should be stored in the NanoClass directory.
Each sample should have a unique sample id.
If there are multiple fastq.gz files per sample they should first be concatenated and added as a single line in the sample.csv.
All samples analyzed together should have the same run ID.
Sample names can differ from fastq.gz file names.
Only numbers and letters are allowed for sample and run labels.
The barcode column should be left empty as NanClass can only take demultiplexed input data.</p>
</dd>
<dt><em>methods</em> [array] (default: [“blastn”,”centrifuge”,”dcmegablast”,”idtaxa”,”kraken”,”mapseq”,”megablast”,”minimap”,”mothur”,”qiime”,”rdp”,”spingo”])</dt><dd><p>Array of tools used in NanoClass run.
Notation should be [“toolname”], including square brackets, even if only a single tool is selected.</p>
</dd>
</dl>
<div class="section" id="porechop">
<h3>porechop<a class="headerlink" href="#porechop" title="Permalink to this headline"></a></h3>
<p>Porechop is a tool to find and remove adapters from Oxford Nanopore reads (<a class="reference external" href="https://github.com/rrwick/Porechop">https://github.com/rrwick/Porechop</a>).
Adapters on the ends of reads are trimmed off, and when a read has an adapter in its middle, it is treated as chimeric and chopped into separate reads.
NanoClass runs Porechop on the raw read files provided by the user.</p>
<p>Porechop is implemented in NanoClass with the following flexible parameters for the user to change:</p>
<dl class="simple">
<dt><em>checkreads</em> [integer] (default: 20000)</dt><dd><p>The number of reads porechop should check to identify adapter sequences.
The adapters identified based on this subset will trimmed from all sequences.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/preprocess.yml”)</dt><dd><p>The recipe for conda to create the preprocessing environment including all dependencies needed to run porechop.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by porechop. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 4000)</dt><dd><p>Memory reserved for porechop per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="nanofilt">
<h3>nanofilt<a class="headerlink" href="#nanofilt" title="Permalink to this headline"></a></h3>
<p>Nanofilt is a tool to trim and filter Oxford Nanopore reads based on length and quality (<a class="reference external" href="https://github.com/wdecoster/nanofilt">https://github.com/wdecoster/nanofilt</a>).
Length filtering can remove spurious sequences that strongly deviate from the expected length of the marker gene of interest.
NanoClass runs Nanofilt on the adapter filtered reads processed by Porechop.</p>
<p>Nanofilt is implemented in NanoClass with the following flexible parameters for the user to change:</p>
<dl class="simple">
<dt><em>minlen</em> [integer] (default: 1)</dt><dd><p>Reads that are shorter than <em>minlen</em> will be discarded by nanofilt. The default value of 1 disables filtering of short reads.
Appropriate value depends on the length range of the marker gene of interest (for most 16S rRNA-based projects is 1400 reasonable).
Users should carefully choose this value to prevent spurious sequences (if too low) or valid data loss (if too high).</p>
</dd>
<dt><em>maxlen</em> [integer] (default: 10000)</dt><dd><p>Reads that are longer than <em>maxlen</em> will be discarded by nanofilt. The default value of 10000 effectively disables filtering of long reads.
Appropriate value depends on the length range of the marker gene of interest (for most 16S rRNA-based projects is 1600 reasonable).
Users should carefully choose this value to prevent spurious sequences (if too high) or valid data loss (if too low).</p>
</dd>
<dt><em>quality</em> [integer] (default: 0)</dt><dd><p>Reads with an average read quality score below <em>quality</em> will be discarded. The default value of 0 disables quality filtering of reads.
Increasing <em>quality</em> goes at the expense of the number of reads retained for downstream analyses.
An appropriate value for a typical Oxford Nanopore MinION dataset is 9 or 10.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/preprocess.yml”)</dt><dd><p>The recipe for conda to create the preprocessing environment including all dependencies needed to run nanofilt.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 2000)</dt><dd><p>Memory reserved for nanofilt per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="subsample">
<h3>subsample<a class="headerlink" href="#subsample" title="Permalink to this headline"></a></h3>
<p>Subsampling can be usefull during tool selection (see typical workflow section of this documentation), when there is more data then necessary, or when the number of reads differs strongly between samples.
If enabled, NanoClass performs subsamplng on the reads that were filtered by Nanofilt.
Subsampling is implemented in NanoClass using seqtk (<a class="reference external" href="https://github.com/lh3/seqtk">https://github.com/lh3/seqtk</a>), with the following flexible parameters for the user to change:</p>
<dl class="simple">
<dt><em>skip</em> [boolian] (default: false)</dt><dd><p>Whether or not the sequence files should be subsampled. Subsampling of reads can substantially decrease runtime but of course only results in the analyses of a subset of the data.</p>
</dd>
<dt><em>samplesize</em> [integer] (default: 10000)</dt><dd><p>The number of reads that will be subsampled. <em>samplesize</em> is ignored when <em>skip</em> is true. Subsampling is applied after filtering by nanofilt.
For tool selection, a <em>samplesize</em> as small as 100 already produced representive tool performance statistics during tests on communities of variable complexity</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/preprocess.yml”)</dt><dd><p>The recipe for conda to create the preprocessing environment including all dependencies needed to run subsample.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
</dl>
</div>
<div class="section" id="nanostats">
<h3>nanostats<a class="headerlink" href="#nanostats" title="Permalink to this headline"></a></h3>
<p>Nanostat computes various statistics for Oxford Nanopore reads (<a class="reference external" href="https://github.com/wdecoster/nanostat">https://github.com/wdecoster/nanostat</a>)
NanoClass runs Nanostat on the filtered reads generated by Nanofilt.
After running NanoClass, the Nanostat results can be found in stats/&lt;run&gt;/nanofilt/, where &lt;run&gt; is the run label provided by the user in the sample.csv.
Nanostat is implementen in NanoClass with the following flexible parameters for the user to change:</p>
<dl class="simple">
<dt><em>environment</em> [string] (default: “../envs/preprocess.yml”)</dt><dd><p>The recipe for conda to create the preprocessing environment including all dependencies needed to run Nanostat.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>threads</em> [interger] (default: 2)</dt><dd><p>The number of threads used per sample by nanostat. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 8000)</dt><dd><p>Memory reserved for nanostat per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="nanoplot">
<h3>nanoplot<a class="headerlink" href="#nanoplot" title="Permalink to this headline"></a></h3>
<p>NanoClass uses pistes to generate quality plots of the Nanofilt-filtered reads (<a class="reference external" href="https://github.com/mbhall88/pistis">https://github.com/mbhall88/pistis</a>).
These plots are included in the NanoClass report. Separate PDF’s can also be found in plots/&lt;run&gt;/nanofilt/, , where &lt;run&gt; is the run label provided by the user in the sample.csv.
Nanoplot is implementen in NanoClass with the following flexible parameters for the user to change:</p>
<dl class="simple">
<dt><em>downsample</em> [integer] (default: 0)</dt><dd><p>Down-sample the sequence files used to generate quality plots to a given number of reads. Set to 0 for no down-sampling.
Down-sampling may decrease runtime somewhat, but as this step does not take long in general it is probably not necessary for most users.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/preprocess.yml”)</dt><dd><p>The recipe for conda to create the preprocessing environment including all dependencies needed to run nanoplot.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 4000)</dt><dd><p>Memory reserved for nanoplot per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="common">
<h3>common<a class="headerlink" href="#common" title="Permalink to this headline"></a></h3>
<p>The parameter settings listed under common are used by several different processes during a typical NanoClass run.
The following common parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>dburl</em> [string] (default:”<a class="reference external" href="https://www.arb-silva.de/fileadmin/silva_databases/qiime/Silva_132_release.zip">https://www.arb-silva.de/fileadmin/silva_databases/qiime/Silva_132_release.zip</a>)</dt><dd><p>The URL to the database resources used by most taxonomic classification tools implemented in NanoClass.
The default url can be used for the analyses of both 16S and 18S rRNA sequences.
For further details on how to change and customize databases see the database section of this documentation.</p>
</dd>
<dt><em>ssu</em> [string] (default: “16S”)</dt><dd><p>Which SILVA database to use. Allowed values: “16S” or “18S”. For further details on how to change and customize databases see the database section of this documentation.</p>
</dd>
<dt><em>group-by</em> [string] (default: “sample”)</dt><dd><p>Whether the taxonomic bar plots included in the report should be grouped by sample or by method (i.e. tool).
For tool selection, choose “method” for classification enter “sample”. See also the typical workflow section of this documentation.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/R4.0.yml”)</dt><dd><p>The recipe for conda to create the R4.0 environment including all dependencies needed to run common processes.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 4000)</dt><dd><p>Memory reserved for building and downloading the common database in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="blastn">
<h3>blastn<a class="headerlink" href="#blastn" title="Permalink to this headline"></a></h3>
<p>BLASTn is one of the 11 classification tools implemented in NanoClass.
BLASTn finds regions of similarity between the Nanofilt-filtered (and, if enabled, subsampled) reads and the entries in the database.
Typically, BLASTn finds multiple good hits to different taxa in the database.
To obtain a consensus classification, NanoClass uses a (LCA) Last Common Ancestor approach.</p>
<p>The following blastn parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>lcaconsensus</em> [float] (default: 0.5)</dt><dd><p>Threshold for calling a consensus. Proportion of blastn hits with the same classification needed to return a consensus.
Taxonomic classifications at each of the 6 taxonomic levels (Domain, Phylum, Class, Order, Family, Genus) will only be assigned if there is consensus at that level.
Range: 0.5-0.99, where 0.5 indicates majority consensus and 0.99 indicates absolute consensus.</p>
</dd>
<dt><em>evalue</em> [float] (default: 0.00001)</dt><dd><p>Expectation value (E) threshold for saving blastn hits. Needs to be provided as decimal number, not in scientific notation!
Decreasing the <em>evalue</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>pctidentity</em> [integer] (default: 80)</dt><dd><p>Percent identity threshold for saving blastn hits. Range: 0-100.
Increasing the <em>pctidentity</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>alnlength</em> [integer] (default: 100)</dt><dd><p>Minimal absolute alignment length for saving blastn hits.
Increasing the <em>alnlength</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>ntargetseqs</em> [integer] (Default: 50)</dt><dd><p>Maximum number of aligned sequences to keep. Increasing the <em>ntargetseqs</em> increases runtime but may also result in more accurate classifications.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/blast.yml”)</dt><dd><p>The recipe for conda to create the blast environment including all dependencies needed to run blastn.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>threads</em> [interger] (default: 10)</dt><dd><p>The number of threads used per sample by blastn. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 500)</dt><dd><p>Memory reserved for blastn per sample per thread in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="dcmegablast">
<h3>dcmegablast<a class="headerlink" href="#dcmegablast" title="Permalink to this headline"></a></h3>
<p>dcMegablast is one of the 11 classification tools implemented in NanoClass.
dcMegablast finds regions of similarity between the Nanofilt-filtered (and, if enabled, subsampled) reads and the entries in the database.
Typically, dcMegablast finds multiple good hits to different taxonomic groups.
To obtain a consensus classification, NanoClass uses a Last Common Ancestor approach.</p>
<p>The following blastn parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>lcaconsensus</em> [float] (default: 0.5)</dt><dd><p>Threshold for calling a consensus. Proportion of dcmegablast hits with the same classification needed to return a consensus.
Taxonomic classifications at each of the 6 taxonomic levels (Domain, Phylum, Class, Order, Family, Genus) will only be assigned if there is consensus at that level.
Range: 0.5-0.99, where 0.5 indicates majority consensus and 0.99 indicates absolute consensus.</p>
</dd>
<dt><em>evalue</em> [float] (default: 0.00001)</dt><dd><p>Expectation value (E) threshold for saving dcmegablast hits. Needs to be provided as decimal number, not in scientific notation!
Decreasing the <em>evalue</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>pctidentity</em> [integer] (default: 80)</dt><dd><p>Percent identity threshold for saving dcmegablast hits. Range: 0-100.
Increasing the <em>pctidentity</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>alnlength</em> [integer] (default: 100)</dt><dd><p>Minimal absolute alignment length for saving dcmegablast hits.
Increasing the <em>alnlength</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>ntargetseqs</em> [integer] (Default: 50)</dt><dd><p>Maximum number of aligned sequences to keep. Increasing the <em>ntargetseqs</em> increases runtime but may also result in more accurate classifications.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/blast.yml”)</dt><dd><p>The recipe for conda to create the blast environment including all dependencies needed to run dcmegablast.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 3000)</dt><dd><p>Memory reserved to build the dcmegablast database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.
The same database is used by blastn and mageblast, if one is already present, it will not be build again.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by dcmegablast. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 50000)</dt><dd><p>Memory reserved for dcmegablast per sample per thread in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="centrifuge">
<h3>centrifuge<a class="headerlink" href="#centrifuge" title="Permalink to this headline"></a></h3>
<p>Centrifuge is one of the 11 classification tools implemented in NanoClass.
In addition to the common database, centrifuge requires mapping and sequence files which will be automatically downloaded by NanoClass using the following parameter settings.
Note that the Centrifuge SILVA version should correspond to the common SILVA version.</p>
<p>The following centrifuge parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>taxmapurl</em> [string] (default: “<a class="reference external" href="https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/taxonomy/taxmap_embl_ssu_ref_nr99_132.txt.gz">https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/taxonomy/taxmap_embl_ssu_ref_nr99_132.txt.gz</a>”)</dt><dd><p>URL to SILVA taxmap. If changed make sure it corresponds to the database provided under common.</p>
</dd>
<dt><em>sequrl</em> [string]: (default: “<a class="reference external" href="https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/SILVA_132_SSURef_Nr99_tax_silva.fasta.gz">https://www.arb-silva.de/fileadmin/silva_databases/release_132/Exports/SILVA_132_SSURef_Nr99_tax_silva.fasta.gz</a>”)</dt><dd><p>URL to SILVA sequences. If changed make sure it corresponds to the database provided under common.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/centrifuge.yml”)</dt><dd><p>The recipe for conda to create the centrifuge environment including all dependencies needed to run centrifuge.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbthreads</em> [interger] (default: 4)</dt><dd><p>The number of threads used by centrifuge to download and build the centrifuge database.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 500)</dt><dd><p>Memory reserved to download and build the centrifuge database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>threads</em> [interger] (default: 1)</dt><dd><p>The number of threads used per sample by centrifuge. Centrifuge is super fast so there is no need to increase <em>threads</em>.</p>
</dd>
<dt><em>memory</em> [integer] (default: 16000)</dt><dd><p>Memory reserved for centrifuge per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="idtaxa">
<h3>idtaxa<a class="headerlink" href="#idtaxa" title="Permalink to this headline"></a></h3>
<p>IDtaxa is one of the 11 classification tools implemented in NanoClass (<a class="reference external" href="https://rdrr.io/bioc/DECIPHER/man/IdTaxa.html">https://rdrr.io/bioc/DECIPHER/man/IdTaxa.html</a>).
IDtaxa is implemented in the DECIPHER R package and classifies sequences by assigning a confidence to taxonomic labels for each taxonomic level.
The training set used by IDtaxa is produced by learntaxa.</p>
<p>The following idtaxa parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>pctthreshold</em> [integer] (default: 60)</dt><dd><p>The confidence at which to truncate the output taxonomic classifications. Lower values of <em>threshold</em> will classify deeper into the taxonomic tree at the expense of accuracy.
Test runs showed that idtaxa failed to produce any taxonomic classifications at the default <em>pctthreshold</em>. Lowering <em>pctthreshold</em> resulted in many spurious classifications.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/R4.0.yml”)</dt><dd><p>The recipe for conda to create R4.0 environment including all dependencies needed to run idtaxa.
If not already present, the environment will be automatically be installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 3000)</dt><dd><p>Memory reserved to train the classifier using learntaxa in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>threads</em> [interger] (default: 8)</dt><dd><p>The number of threads used per sample by idtaxa. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 10000)</dt><dd><p>Memory reserved for idtaxa per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="kraken">
<h3>kraken<a class="headerlink" href="#kraken" title="Permalink to this headline"></a></h3>
<p>Kraken2 is one of the 11 classification tools implemented in NanoClass (<a class="reference external" href="https://github.com/DerrickWood/kraken2/wiki/Manual">https://github.com/DerrickWood/kraken2/wiki/Manual</a>).
Unlike the other tools, Kraken2 is a k-mer based classifier with an integrated lowest common ancestor (LCA) approach and is super fast.</p>
<p>The following kraken parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>dbtype</em> [string] (default: “silva”)</dt><dd><p>Kraken supports 16S rRNA databases not based on NCBI’s taxonomy, including Greengenes, SILVA, and RDP. Only SILVA was tested to use within NanoClass.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/kraken2.yml”)</dt><dd><p>The recipe for conda to create the kraken2 environment including all dependencies needed to run kraken.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbthreads</em> [interger] (default: 8)</dt><dd><p>The number of threads used by kraken to download and build the kraken database.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 1000)</dt><dd><p>Memory reserved to download and build the kraken database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by kraken.</p>
</dd>
<dt><em>memory</em> [integer] (default: 500)</dt><dd><p>Memory reserved for kraken per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="megablast">
<h3>megablast<a class="headerlink" href="#megablast" title="Permalink to this headline"></a></h3>
<p>Megablast is one of the 11 classification tools implemented in NanoClass.
Megablast finds regions of similarity between the Nanofilt-filtered (and, if enabled, subsampled) reads and the entries in the database.
Typically, Megablast finds multiple good hits to different taxonomic groups.
To obtain a consensus classification, NanoClass uses a Last Common Ancestor approach.</p>
<p>The following blastn parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>lcaconsensus</em> [float] (default: 0.5)</dt><dd><p>Threshold for calling a consensus. Proportion of megablast hits with the same classification needed to return a consensus.
Taxonomic classifications at each of the 6 taxonomic levels (Domain, Phylum, Class, Order, Family, Genus) will only be assigned if there is consensus at that level.
Range: 0.5-0.99, where 0.5 indicates majority consensus and 0.99 indicates absolute consensus.</p>
</dd>
<dt><em>evalue</em> [float] (default: 0.00001)</dt><dd><p>Expectation value (E) threshold for saving megablast hits. Needs to be provided as decimal number, not in scientific notation!
Decreasing the <em>evalue</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>pctidentity</em> [integer] (default: 80)</dt><dd><p>Percent identity threshold for saving megablast hits. Range: 0-100.
Increasing the <em>pctidentity</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>alnlength</em> [integer] (default: 100)</dt><dd><p>Minimal absolute alignment length for saving megablast hits.
Increasing the <em>alnlength</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organisms that are poorly represented in the database.</p>
</dd>
<dt><em>ntargetseqs</em> [integer] (Default: 50)</dt><dd><p>Maximum number of aligned sequences to keep. Increasing the <em>ntargetseqs</em> increases runtime but may also result in more accurate classifications.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/blast.yml”)</dt><dd><p>The recipe for conda to create the blast environment including all dependencies needed to run megablast.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 3000)</dt><dd><p>Memory reserved to build the megablast database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.
The same database is used by blastn and mageblast, if one is already present, it will not be build again.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by megablast. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 50000)</dt><dd><p>Memory reserved for megablast per sample per thread in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
<div class="section" id="minimap">
<h3>minimap<a class="headerlink" href="#minimap" title="Permalink to this headline"></a></h3>
<p>Minimap2 is one of the 11 classification tools implemented in NanoClass.
Minimap2 is a fast sequence alignment program used to align noisy Oxfort Nanopore MinION reads against a reference database (<a class="reference external" href="https://github.com/lh3/minimap2">https://github.com/lh3/minimap2</a>).
In NanoClass, Minimap2 aligns Nanofilt-filtered (and, if enabled, subsampled) reads against the reference database, where the user can determine how many seconday alignments to consider.
To obtain a consensus classification of the primary and seconday alignments, NanoClass uses a Last Common Ancestor approach</p>
<p>The following minimap parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>lcaconsensus</em> [float] (default: 0.5)</dt><dd><p>Threshold for calling a consensus. Proportion of minimap primary and secondary alignments with the same classification needed to return a consensus.
Taxonomic classifications at each of the 6 taxonomic levels (Domain, Phylum, Class, Order, Family, Genus) will only be assigned if there is consensus at that level.
Range: 0.5-0.99, where 0.5 indicates majority consensus and 0.99 indicates absolute consensus.</p>
</dd>
<dt><em>ntargetseqs</em> [integer] (default: 10)</dt><dd><p>Number of secondary alignments to save and consider when computing LCA.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/minimap2.yml”)</dt><dd><p>The recipe for conda to create the minimap2 environment including all dependencies needed to run minimap.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 3000)</dt><dd><p>Memory reserved for minimap per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by minimap. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
</dl>
</div>
<div class="section" id="mothur">
<h3>mothur<a class="headerlink" href="#mothur" title="Permalink to this headline"></a></h3>
<p>Mothur is one of the 11 classification tools implemented in NanoClass.
In NanoClass, Mother aligns the Nanofilt-filtered (and, if enabled, subsampled) reads against the reference database
using the align_seqs function with ksize=6 and align=needleman.</p>
<p>Because mothur saves the entire database in memory, the memory requirements are high which means it may not be able to run it on a standard desktop or laptop computer.
In that case, just skip mothur in the NanoClass run by removing it from the methods array at the top of the config.yaml.</p>
<p>The following mothur parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>environment</em> [string] (default: “../envs/mothur.yml”)</dt><dd><p>The recipe for conda to create the mothur environment including all dependencies needed to run mothur.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 1000)</dt><dd><p>Memory reserved for mothur per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
<dt><em>threads</em> [interger] (default: 8)</dt><dd><p>The number of threads used per sample by mothur. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 5000)</dt><dd><p>Memory reserved to build the mothur database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
</dl>
</div>
<div class="section" id="qiime">
<h3>qiime<a class="headerlink" href="#qiime" title="Permalink to this headline"></a></h3>
<p>QIIME2 is one of the 11 classification tools implemented in NanoClass.
NanoClass uses the QIIME2 feature-classifier classify-consensus-vsearch utility to classify the Nanofilt-filtered (and, if enabled, subsampled) reads using the reference database.</p>
<p>The following qiime parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>lcaconsensus</em> [float] (default: 0.51)</dt><dd><p>Threshold for calling a consensus. Proportion of qiime hits with the same classification needed to return a consensus.
Taxonomic classifications at each of the 6 taxonomic levels (Domain, Phylum, Class, Order, Family, Genus) will only be assigned if there is consensus at that level.
Range: 0.51-1, where 0.51 indicates majority consensus and 1 indicates absolute consensus.</p>
</dd>
<dt><em>pctidentity</em> [float] (default: 0.8)</dt><dd><p>Proportion identity threshold for saving minimap hits. Range: 0-1. Should be given as a proportion, not a percentage.
Increasing the <em>pctidentity</em> may result in more consensus classifications but could also reduce the likelyhood of finding any taxonomic classification, in particular for organism$</p>
</dd>
<dt><em>ntargetseqs</em> [integer] (Default: 10)</dt><dd><p>Maximum number of aligned sequences to keep. Increasing the <em>ntargetseqs</em> increases runtime but may also result in more accurate classifications.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/qiime2.yml”)</dt><dd><p>The recipe for conda to create the qiime2 environment including all dependencies needed to run qiime.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 10000)</dt><dd><p>Memory reserved for qiime per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by qiime2. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 3000)</dt><dd><p>Memory reserved to build the qiime database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
</dl>
</div>
<div class="section" id="rdp">
<h3>rdp<a class="headerlink" href="#rdp" title="Permalink to this headline"></a></h3>
<p>RDP is one of the 11 classification tools implemented in NanoClass.
NanoClass runs the assignTaxonomy utility of the DADA2 R package on the Nanofilt-filtered (and, if enabled, subsampled) reads using the reference database.
It implements the RDP classifier algorithm with kmer size 8 and 100 bootstrap replicates and assignes taxonomy based on he minimum bootstrap confidence.</p>
<p>The following rdp parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>pctthreshold</em> [integer] (default: 60)</dt><dd><p>The minimum bootstrap confidence at which to truncate the output taxonomic classifications.
Lower values of <em>pctthreshold</em> will classify deeper into the taxonomic tree at the expense of accuracy.</p>
</dd>
<dt><em>environment</em> [string] (default: “../envs/R4.0.yml”)</dt><dd><p>The recipe for conda to create the R4.0 environment including all dependencies needed to run rdp.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>memory</em> [integer] (default: 5000)</dt><dd><p>Memory reserved for rdp per sample in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
<dt><em>threads</em> [interger] (default: 8)</dt><dd><p>The number of threads used per sample by rdp. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 75000)</dt><dd><p>Memory reserved to build the rdp database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
</dl>
</div>
<div class="section" id="spingo">
<h3>spingo<a class="headerlink" href="#spingo" title="Permalink to this headline"></a></h3>
<p>Spingo is one of the 11 classification tools implemented in NanoClass (<a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0747-1">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0747-1</a>).
NanoClass runs Spingo on the Nanofilt-filtered (and, if enabled, subsampled) reads using the reference database.</p>
<p>The following spingo parameter settings can be changed by the user:</p>
<dl class="simple">
<dt><em>environment</em> [string] (default: “../envs/spingo.yml”)</dt><dd><p>The recipe for conda to create the spingo environment including all dependencies needed to run spingo.
If not already present, the environment will be automatically installed by NanoClass when using the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span></code> option.
Changes to the environment are not recommended unless you are an advanced user.</p>
</dd>
<dt><em>dbthreads</em> [interger] (default: 16)</dt><dd><p>The number of threads used to build the spingo database.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>dbmemory</em> [integer] (default: 50000)</dt><dd><p>Memory reserved to build the spingo database in Mb.
This step is independent of the properties of the data to be analysed and therefore does not need to be changed.</p>
</dd>
<dt><em>threads</em> [interger] (default: 16)</dt><dd><p>The number of threads used per sample by spingo. Increasing the number of threads may speed up the analyses.
NanoClass will automatically downgrade the number of threads when fewer cores are specified by the user in the <code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--use-conda</span> <span class="pre">--cores</span> <span class="pre">&lt;ncores&gt;</span></code>
command. It is therefore not necesary to change this value for a typical run of NanoClass.</p>
</dd>
<dt><em>memory</em> [integer] (default: 50000)</dt><dd><p>Memory reserved for spingo per sample per thread in Mb. If insufficient, NanoClass will automatically double the amount of memory and try again.
Therefore, it is not necessary to change this value for a typical NanoClass run.</p>
</dd>
</dl>
</div>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="databases.html" class="btn btn-neutral float-left" title="Databases" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Evelien Jongepier.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>