

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Taxonomic classification &mdash; QIIME2-workshop 2021.06 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Metabarcoding analyses on Crunchomics" href="crunchomics.html" />
    <link rel="prev" title="Quality control and feature tabulation" href="denoise.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> QIIME2-workshop
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="genintro.html">General Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="denoise.html">Quality control and feature tabulation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Taxonomic classification</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#importing-the-database">Importing the database</a></li>
<li class="toctree-l2"><a class="reference internal" href="#extract-reference-reads">Extract reference reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="#train-the-classifier">Train the classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1">Taxonomic classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="#taxonomic-barplot">Taxonomic barplot</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="crunchomics.html">Metabarcoding analyses on Crunchomics</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">QIIME2-workshop</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Taxonomic classification</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="taxonomic-classification">
<h1>Taxonomic classification<a class="headerlink" href="#taxonomic-classification" title="Permalink to this headline">¶</a></h1>
<p>In this second part of the workshop we will continue with the taxonomic classification of the representative sequences from the DADA2-denoising performed yesterday.</p>
<p>QIIME 2 provides several methods to predict the most likely taxonomic affiliation of your features, including both alignment-based consensus methods and Naive Bayes (and other machine-learning) methods.</p>
<p>Here, we will use a Naive Bayes classifier, which must be trained on taxonomically defined reference sequences covering the target region of interest.
This tutorial addresses the following questions:</p>
<ol class="arabic simple">
<li><p>How to import the SILVA 16S rRNA taxonomic database as a reference in QIIME2?</p></li>
<li><p>How to extract the fragments from the SILVA reference sequences corresponding to the primers that were used?</p></li>
<li><p>How to train a custom classifier on these particular fragments of the 16S rRNA gene?</p></li>
<li><p>How to use the trained classifier to get a taxonomic classification of the representative sequences?</p></li>
<li><p>How to vizualise the taxonomic classification as interactive barplots?</p></li>
</ol>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Training a classifier takes up quite a bit of RAM and time.
Therefore, I have pre-computed the classifier for you.
I recommend you just look through the next steps in the analyses without actually running them.</p>
</div>
<p>See the QIIME2 <a class="reference external" href="https://docs.qiime2.org/2020.8/tutorials/feature-classifier/">Training feature classifiers with q2-feature-classifier</a> tutorial further information.</p>
<div class="section" id="importing-the-database">
<h2>Importing the database<a class="headerlink" href="#importing-the-database" title="Permalink to this headline">¶</a></h2>
<p>Two elements are required for training the classifier: the reference sequences and the corresponding taxonomic classifications.
In the db directory of your data package, you will find these two files for the 16S rRNA SILVA data base release version 138, at 99% sequence similarity cutoff.
We can import these files into QIIME2 as artifacts, similar to how you imported data yesterday:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --type <span class="s2">&quot;FeatureData[Sequence]&quot;</span> <span class="se">\</span>
  --input-path db/SILVA_138_99_16S-ref-seqs.fna <span class="se">\</span>
  --output-path db/SILVA_138_99_16S-ref-seqs.qza
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --type <span class="s2">&quot;FeatureData[Taxonomy]&quot;</span> <span class="se">\</span>
  --input-format HeaderlessTSVTaxonomyFormat <span class="se">\</span>
  --input-path db/SILVA_138_99_16S-ref-taxonomy.txt <span class="se">\</span>
  --output-path db/SILVA_138_99_16S-ref-taxonomy.qza
</pre></div>
</div>
<p>Lets have a look at the first entry in the database, like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head -2 db/SILVA_138_99_16S-ref-seqs.fna
</pre></div>
</div>
<p>Which gives the following output:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;CP013078.2406498.2408039
AGAGATTGAACTGAAGAGTTTGATCCTGGCTCAGATTGAACGCTGGCGGGATGCTTTACACATGCAAGTCGGACGGCAGCACGGGCTTCGGCCTGGTGGCGAGTGGCGAACGGGTGAGTAATGTATCGGAACGTGCCCAGTAGCGGGGGATAACTACGCGAAAGCGTGGCTAATACCGCATACGCCCTACGGGGGAAAGCGGGGGACCTTCGGGCCTCGCACTATTGGAGCGGCCGATATCGGATTAGCTAGTTGGTGGGGTAACGGCCTACCAAGGCGACGATCCGTAGCTGGTTTGAGAGGACGACCAGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTGGGGAATTTTGGACAATGGGGGCAACCCTGATCCAGCCATCCCGCGTGTGCGATGAAGGCCTTCGGGTTGTAAAGCACTTTTGGCAGGAAAGAAACGGCACGGGCTAATATCCTGTGCAACTGACGGTACCTGCAGAATAAGCACCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTCGGAAAGAAAGATGTGAAATCCCAGGGCTTAACCTTGGAACTGCATTTTTAACTACCGGGCTAGAGTGTGTCAGAGGGAGGTGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAGGAACACCGATGGCGAAGGCAGCCTCCTGGGATAACACTGACGCTCATGCACGAAAGTGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCCTAAACGATGTCAACTAGCTGTTGGGGCCTTCGGGCCTTGGTAGCGCAGCTAACGCGTGAAGTTGACCGCCTGGGGAGTACGGTCGCAAGATTAAAACTCAAAGGAATTGACGGGGACCCGCACAAGCGGTGGATGATGTGGATTAATTCGATGCAACGCGAAAAACCTTACCTACCCTTGACATGTCTGGAATCCCGAAGAGATTTGGGAGTGCTCGCAAGAGAACCGGAACACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCATTAGTTGCTACGAAAGGGCACTCTAATGAGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCCTCATGGCCCTTATGGGTAGGGCTTCACACGTCATACAATGGTCGGGACAGAGGGTTGCCAACCCGCGAGGGGGAGCCAATCCCAGAAACCCGGTCGTAGTCCGGATCGCAGTCTGCAACTCGACTGCGTGAAGTCGGAATCGCTAGTAATCGCGGATCAGCATGTCGCGGTGAATACGTTCCCGGGTCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTTTACCAGAAGTAGTTAGCCTAACCGCAAGGGGGGCGATTACCACGGTAGGATTCATGACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTTAAGA
</pre></div>
</div>
<p>Lets also check the associated 7-level taxonomy for this database entry:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>grep CP013078.2406498.2408039 db/SILVA_138_99_16S-ref-taxonomy.txt
</pre></div>
</div>
<p>And the output:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>CP013078.2406498.2408039     D_0__Bacteria<span class="p">;</span>D_1__Proteobacteria<span class="p">;</span>D_2__Gammaproteobacteria<span class="p">;</span>D_3__Burkholderiales<span class="p">;</span>D_4__Alcaligenaceae<span class="p">;</span>D_5__Bordetella<span class="p">;</span>D_6__Bordetella pertussis
</pre></div>
</div>
<div class="admonition-question-11 admonition">
<p class="admonition-title">Question 11</p>
<p>How long was the target sequence again? How does that compare the the sequence length in the SILVA database?
What are two major disadvantages of using full length 16S rRNA gene in your taxonomic classification?</p>
</div>
</div>
<div class="section" id="extract-reference-reads">
<h2>Extract reference reads<a class="headerlink" href="#extract-reference-reads" title="Permalink to this headline">¶</a></h2>
<p>The 16S rRNA gene is characterized by both hyper variable and very conserved regions.
Taxonomic classification accuracy of 16S rRNA gene sequences improves when a Naive Bayes classifier is trained
on only the region of the target sequences that was sequenced
(<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/21716311/">Werner et al. 2012</a>).</p>
<p>From the trimming step yesterday, we know which primer sequences were used.
We can now use these same sequences to extract the corresponding region of the 16S rRNA sequences from the SILVA database, like so:</p>
<p>Note, this takes a while even using 16 cpus like I did.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-classifier extract-reads <span class="se">\</span>
  --i-sequences db/SILVA_138_99_16S-ref-seqs.qza <span class="se">\</span>
  --p-f-primer GTGYCAGCMGCCGCGGTAA <span class="se">\</span>
  --p-r-primer CCGYCAATTYMTTTRAGTTT <span class="se">\</span>
  --o-reads db/SILVA_138_99_16S-ref-frags.qza <span class="se">\</span>
  --p-n-jobs <span class="m">16</span>
</pre></div>
</div>
<p>Lets have a look at the result:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data db/SILVA_138_99_16S-ref-frags.qza <span class="se">\</span>
  --o-visualization db/SILVA_138_99_16S-ref-frags.qzv

qiime tools view db/SILVA_138_99_16S-ref-frags.qzv
</pre></div>
</div>
<div class="admonition-question-12 admonition">
<p class="admonition-title">Question 12</p>
<p>How does the sequence length distribution of the reference fragments compare to that of the representative sequences?</p>
</div>
</div>
<div class="section" id="train-the-classifier">
<h2>Train the classifier<a class="headerlink" href="#train-the-classifier" title="Permalink to this headline">¶</a></h2>
<p>In this step we use the reference fragments you just created to train your classifier specifically on your region of interest.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>You should only run this step if you have &gt;32GB of RAM available!</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-classifier fit-classifier-naive-bayes <span class="se">\</span>
  --i-reference-reads db/SILVA_138_99_16S-ref-frags.qza <span class="se">\</span>
  --i-reference-taxonomy db/SILVA_138_99_16S-ref-taxonomy.qza <span class="se">\</span>
  --o-classifier db/SILVA_138_99_16S-ref-classifier.qza
</pre></div>
</div>
<p>Please note that this classifier is not very specific with respect to which environment the samples come from,
because it assumes that all species in the reference database are equally likely to be observed in your samples
(i.e., that sea-floor microbes are just as likely to be found in a stool sample as microbes usually associated with stool).</p>
<p>It is actually possible to incorporate environment-specific taxonomic abundance information to improve species inference.
This bespoke method has been shown to improve classification accuracy when compared to traditional Naive Bayes classifiers
(<a class="reference external" href="https://www.nature.com/articles/s41467-019-12669-6">Kaehler et al. 2019</a>).</p>
<p>To train a classifier using this bespoke method, we need to provide an additional file with taxonomic weigths
(see <code class="docutils literal notranslate"><span class="pre">--i-class-weight</span></code> in the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">feature-classifier</span> <span class="pre">fit-classifier-naive-bayes</span></code> help function)
Pre-assembled taxonomic weights can be found in the readytowear collection at <a class="reference external" href="https://github.com/BenKaehler/readytowear">https://github.com/BenKaehler/readytowear</a>.
I cannot judge how well they fit your particular environment, so be very, very careful in using them unless you know what you are doing.</p>
<p>Here, we will continue with the ‘naive’ Naive Bayes classifier we just created.</p>
</div>
<div class="section" id="id1">
<h2>Taxonomic classification<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>So now we have a trained classifier and a set of representative sequences from your DADA2-denoise analyses.
Lets run it and find out which microbes were present in your samples.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>I ran this step on a computational cluster because it requires ~50 GB of RAM.
Don’t run this yourself unless you have lots of RAM on your system.</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-classifier classify-sklearn <span class="se">\</span>
  --i-classifier db/SILVA_138_99_16S-ref-classifier.qza <span class="se">\</span>
  --p-n-jobs <span class="m">16</span> <span class="se">\</span>
  --i-reads dada2/dada2-reprseqs.qza <span class="se">\</span>
  --o-classification taxonomy/dada2-SILVA_138_99_16S-taxonomy.qza
</pre></div>
</div>
<p>You can view the taxonomic annotation of each of the representative sequences like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
 --m-input-file taxonomy/dada2-SILVA_138_99_16S-taxonomy.qza <span class="se">\</span>
 --o-visualization taxonomy/dada2-SILVA_138_99_16S-taxonomy.qzv

qiime2 tools view taxonomy/dada2-SILVA_138_99_16S-taxonomy.qzv
</pre></div>
</div>
<p>Note that this also reports a confidence score ranging between 0.7 and 1.
The lower limit of 0.7 is the default value (see also the <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">feature-classifier</span> <span class="pre">classify-sklearn</span></code> help function).
You can opt for a lower value to increase the number of features with a classification, but beware that that will also increase the risk of spurious classifcations!</p>
</div>
<div class="section" id="taxonomic-barplot">
<h2>Taxonomic barplot<a class="headerlink" href="#taxonomic-barplot" title="Permalink to this headline">¶</a></h2>
<p>In this final step we will create an interactive barplot, showing the relative abundances at different taxonomic levels for each of the samples.
Before running the command, we will need to prepair a metadata file.
This metadata file should contain information on the samples. For instance, at which depth was the sample taken,
from which location does it come, was it subjected to experimental or control treatment etc. etc.
This information is of course very specific to the study design but at the very least it should look like this (see also <code class="docutils literal notranslate"><span class="pre">data/META.tsv</span></code>):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#SampleID</span>
<span class="c1">#q2:types</span>
DUMMY1
DUMMY10
...
</pre></div>
</div>
<p>but we can add any variables, like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#SampleID    BarcodeSequence Location        depth   location        treatment       grainsize       flowrate        age</span>
<span class="c1">#q2:types    categorical     categorical     categorical     catechorical    categorical     numeric numeric numeric</span>
&lt;your data&gt;
</pre></div>
</div>
<p>See the <a class="reference external" href="https://docs.qiime2.org/2020.8/tutorials/metadata/">Metadata in QIIME 2</a> tutorial for further information.</p>
<p>I created a minimal example for your data here <code class="docutils literal notranslate"><span class="pre">data/META.tsv</span></code>
Just complement it with the specific variables of your study or continue with this minimal example.
Then create and view the barplot vizualisation, like so:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime taxa barplot <span class="se">\</span>
  --i-table dada2/dada2-table.qza <span class="se">\</span>
  --i-taxonomy taxonomy/dada2-SILVA_138_99_16S-taxonomy.qza <span class="se">\</span>
  --m-metadata-file data/META.tsv <span class="se">\</span>
  --o-visualization taxonomy/dada2-SILVA_138_99_16S-taxplot.qzv

qiime tools view taxonomy/dada2-SILVA_138_99_16S-taxplot.qzv
</pre></div>
</div>
<div class="admonition-question-13 admonition">
<p class="admonition-title">Question 13</p>
<p>Which family is the most abundant in your samples, and which class?
Can you see clear differences in relative abundances between your treatment/sample location/depth/…?
… &lt;here is where you get to add your own questions&gt;</p>
</div>
<p>Now that we have come to the end of this workshop, lets have a final look at the
metadata for the taxonomy analyses. Just go to <a class="reference external" href="https://view.qiime2.org/">https://view.qiime2.org/</a>
and load your <cite>artifact</cite> taxonomy/dada2-SILVA_138_99_16S-taxonomy.qza.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="crunchomics.html" class="btn btn-neutral float-right" title="Metabarcoding analyses on Crunchomics" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="denoise.html" class="btn btn-neutral float-left" title="Quality control and feature tabulation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, Evelien Jongepier

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>